client/src/components/AlumniDirectory.js

```diff
@@ -102,32 +115,78 @@ export default class AlumniDirectory extends Component {
     populateSearchDropdownStates(entries) {
         let gradYears = [];
         let allText = [];
+        let topicsArray = [];
+        let topicsSet = new Set();
+        let interestsArray = [];
+        let interestsSet = new Set();
         let display = [];
-        let i = 0;
+        let i = 0, j = 0;
+
         for (let post of entries) {
-            if(!gradYears.find(year => year['value'] === post.gradYear)) {
+            if (!gradYears.find(year => year['value'] === post.gradYear)) {
                 gradYears.push({
                     key: post.gradYear,
                     text: post.gradYear,
                     value: post.gradYear
                 });
             }
+
+            let topicsForAlumnus = post.topics;
```

Comment:
So `for (let varName in arr) { ... }` gives you the index in the iteration body and `for (let varName of arr) { ... }` gives you the actual object in the iteration body...